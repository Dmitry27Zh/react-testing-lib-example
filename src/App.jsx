import { useEffect, useState } from 'react'

function App() {
  const [data, setData] = useState(null)
  const [toggle, setToggle] = useState(false)
  const [value, setValue] = useState('')
  const onClick = () => setToggle((prev) => !prev)
  useEffect(() => {
    setTimeout(() => {
      setData({})
    }, 100)
  }, [])

  return (
    <div>
      <h1 data-testid="value-elem">{value}</h1>
      {toggle && <div data-testid="toggle-elem">toggle</div>}
      {data && <div style={{ color: 'red' }}>data</div>}
      <h2>Hello world</h2>
      <button data-testid="toggle-btn" onClick={onClick}>
        Click me
      </button>
      <input type="text" placeholder="input value..." onChange={(e) => setValue(e.target.value)} />
    </div>
  )
}

export default App
