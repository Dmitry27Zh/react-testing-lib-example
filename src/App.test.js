import { render, screen } from '@testing-library/react'
import App from './App'

describe('TEST_APP', () => {
  test('renders learn react components', () => {
    render(<App />)
    const helloWorldElement = screen.getByText(/hello world/i)
    const btn = screen.getByRole('button')
    const input = screen.getByPlaceholderText(/input value.../i)
    expect(helloWorldElement).toBeInTheDocument()
    expect(btn).toBeInTheDocument()
    expect(input).toMatchSnapshot()
    screen.debug()
  })
  test('screen api', () => {
    render(<App />)
    const helloWorldElem = screen.queryByText(/hello2/i)
    expect(helloWorldElem).toBeNull()
  })
})
