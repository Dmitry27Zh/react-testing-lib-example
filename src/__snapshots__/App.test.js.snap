// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST_APP renders learn react components 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;

exports[`renders learn react components 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;

exports[`renders learn react link 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;
